src/main/java/ch/njol/util/coll/CollectionUtils.java

```diff
@@ -435,5 +437,26 @@ public static Double[] wrap(double[] primitive) {
 		}
 		return wrapped;
 	}
+
+	/**
+	 * Removes the first n elements from a Collection that matches the filter
+	 *
+	 * @param coll the {@link Collection} to remove from.
+	 * @param n the amount of elements to remove
+	 * @param filter the {@link Predicate} / filter that will be checked to remove the elements
+	 * @return amount of elements that matched the filter and have been removed.
+	 */
+	public <E> int removeFirstIf(Collection<E> coll, int n, Predicate<E> filter) {
+		Iterator<E> each = coll.iterator();
+		int removed = 0;
+
+		for (; each.hasNext() && n > 0; n--) {
+			if (filter.test(each.next())) {
+				each.remove();
```

Comment:
Iterator#remove is an optional operation, it is not guaranteed to be available. You should probably switch to removing from the collection directly.